nectar/src/network.rs

```diff
@@ -99,15 +93,15 @@ pub struct SetupSwapContext {
 /// A `NetworkBehaviour` that delegates to the `Orderbook` and `SetupSwap`
 /// behaviours.
 #[derive(NetworkBehaviour)]
-#[behaviour(out_event = "Event", poll_method = "poll")]
+#[behaviour(out_event = "orderbook::BehaviourOutEvent", poll_method = "poll")]
```

Comment:
Use `event_process = false` :)

https://docs.rs/libp2p/0.28.1/libp2p/swarm/trait.NetworkBehaviour.html#deriving-networkbehaviour

Reply:
I don't think it behaves as you expect. `swarm.next()` ends up returning `()`.

Reply:
Here is how it should behave: https://github.com/libp2p/rust-libp2p/blob/9647557f0d81a37f1c2062638aa10eba50d91631/misc/core-derive/tests/test.rs#L267-L308

Is this not what is happening?

Reply:
I'll look in more details but I am not yet sure how it is better than the status quo.

Reply:
You should be get rid of your "events" field and the dedicated "poll" function :)